ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -613,8 +613,9 @@ private boolean isEligibleForCompaction(CompactionInfo ci,
         return false;
       }
 
-      if (isNoAutoCompactSet(t.getParameters())) {
-        LOG.info("Table " + tableName(t) + " marked " + hive_metastoreConstants.TABLE_NO_AUTO_COMPACT +
+      Map<String, String> dbParams = computeIfAbsent(ci.dbname, () -> resolveDatabase(ci)).getParameters();
```

Comment:
`isEligibleForCompaction` accepts the skipDBs SET, we should populate and reuse it when checking the next candidates

Reply:
isNoAutoCompactSet considers both db level and table level configs and so if we skip the db when table level config is set, then it will affect other tables. Also isEligibleForCompaction accepts skipTables SET and skipDBs , skipTables both are not used inside  isNoAutoCompactSet., I am not sure if there is any particular reason for this.

Reply:
````
if (replIsCompactionDisabledForTable(t)) {
        skipTables.add(ci.getFullTableName());
        return false;
}
````
Would be better if we could refactor the method and use a cache of skipTables/skipDBs instead of doing the same evaluation for every Table in skipped Db / every Partition of skipped Table